gtk4/src/builder_rust_scope.rs

```diff
@@ -0,0 +1,143 @@
+use crate::subclass::builder_scope::BuilderScopeImpl;
+use crate::Builder;
+use crate::BuilderClosureFlags;
+use crate::BuilderError;
+use crate::BuilderScope;
+use glib::prelude::*;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::Closure;
+use std::collections::HashMap;
+use std::rc::Rc;
+
+glib::wrapper! {
+    pub struct BuilderRustScope(ObjectSubclass<BuilderRustScopePrivate>)
+        @implements BuilderScope;
+}
+
+#[derive(Default)]
+pub struct BuilderRustScopePrivate {
+    callbacks:
+        std::cell::RefCell<HashMap<String, Rc<dyn Fn(&[glib::Value]) -> Option<glib::Value>>>>,
+}
```

Comment:
```suggestion
pub struct BuilderRustScopePrivate(std::cell::RefCell<HashMap<String, Rc<dyn Fn(&[glib::Value]) -> Option<glib::Value>>>>);
```
Also i would do include RefCell in-tree; create a Type for the GClosures you have here to make this part of code more readable